dask/dataframe/io/parquet/arrow.py

```diff
@@ -1699,40 +1705,69 @@ def _read_table(
         return arrow_table
 
     @classmethod
-    def _arrow_table_to_pandas(
-        cls, arrow_table: pa.Table, categories, use_nullable_dtypes=False, **kwargs
-    ) -> pd.DataFrame:
-        _kwargs = kwargs.get("arrow_to_pandas", {})
-        _kwargs.update({"use_threads": False, "ignore_metadata": False})
-
-        if use_nullable_dtypes:
-            # Determine is `pandas` or `pyarrow`-backed dtypes should be used
-            if use_nullable_dtypes == "pandas":
-                default_types_mapper = PYARROW_NULLABLE_DTYPE_MAPPING.get
+    def _determine_type_mapper(cls, **kwargs):
+        use_nullable_dtypes = kwargs.get("use_nullable_dtypes", False)
+        convert_string = kwargs.get("convert_string", False)
+        user_mapper = kwargs.get("arrow_to_pandas", {}).get("types_mapper")
```

Comment:
Could we include `use_nullable_dtypes`, `convert_string`, and `user_mapper` in the function signature? There is nothing *wrong* with extracting them from `kwargs`, but I'd like to move in the direction of being more explicit about what key-word arguments are expected.

Reply:
Note that I realize the original code was already being pretty sloppy with `kwargs` - I'll try not to let those original mistakes expand the scope of this PR too much :)

Reply:
@rjzamora This way gives more freedom to subclasses to implement their own version of `_determine_type_mapper` if they want, without breaking the contract and making `mypy` unhappy. I did it after realizing that `dask-geopandas` needs their `ArrowDatasetEngine` to be compatible with ours. This function is internal to the class; if it was user-facing, I'd use explicit kwargs. What do you think?

Reply:
I was probably mistaken about `user_mapper`, but I'm not sure I understand how moving `use_nullable_dtypes` or `convert_string` would cause a problem. For example, I'm pretty sure the difference between these cases is purely stylistic:

```python
    def _determine_type_mapper(cls, **kwargs):
        use_nullable_dtypes = kwargs.get("use_nullable_dtypes", False)
        convert_string = kwargs.get("convert_string", False)
        user_mapper = kwargs.get("arrow_to_pandas", {}).get("types_mapper")
        ...
```

```python
    def _determine_type_mapper(cls, use_nullable_dtypes=False, convert_string=False, **kwargs):
        user_mapper = kwargs.get("arrow_to_pandas", {}).get("types_mapper")
        ...
```

Reply:
@rjzamora Someone may want to have type mapping logic that depends on different kwargs and doesn't care about `use_nullable_dtypes` and `convert_string`. Right now this is not the case though, so I guess I don't need to think so far ahead. I'll make them explicit.